src/decorator.js

```diff
@@ -0,0 +1,16 @@
+import shouldPureComponentUpdate from './function';
+import { Component } from 'react';
+
+export default function PureRender(Component) {
+  const componentName = Component.name || 'component';
+
+  if (!(Component.prototype instanceof Component)) {
+    throw new Error(`${componentName} is not a React component`);
```

Comment:
This is not necessary. You don't have to inherit from `Component` for it to work. Maybe replace this with a simple check for prototype's existance?